pep-0665.rst

```diff
@@ -1,79 +1,97 @@
 PEP: 665
-Title: Specifying Installation Requirements for Python Projects
+Title: A file format to list Python dependencies for reproducibility of an application
 Author: Brett Cannon <brett@python.org>,
         Pradyun Gedam <pradyunsg@gmail.com>,
         Tzu-ping Chung <uranusjr@gmail.com>
-PEP-Delegate:
-Discussions-To: https://discuss.python.org/t/pep-665-specifying-installation-requirements-for-python-projects/9911
+PEP-Delegate: Paul Moore <p.f.moore@gmail.com>
+Discussions-To: https://discuss.python.org/t/9911
 Status: Draft
 Type: Standards Track
 Content-Type: text/x-rst
 Created: 29-Jul-2021
-Post-History: 29-Jul-2021
+Post-History: 29-Jul-2021, 03-Nov-2021
 Resolution:
 
 ========
 Abstract
 ========
 
-This PEP specifies a file format to list the Python package
-installation requirements for a project. The list of projects is
-considered exhaustive for the installation target and thus
-*locked down*, not requiring any information beyond the platform being
-installed for and the *lock file* listing the required dependencies
-to perform a successful installation of dependencies.
+This PEP specifies a file format to specify the list of Python package
+installation requirements for an application, and the relation between the
+specified requirements. The list of requirements is considered
+exhaustive for the installation target, and thus not requiring any
+information beyond the platform being installed for, and the file
+itself. The file format is flexible enough to allow installing the
+requirements across different platforms, which guarantees
+reproducibility on multiple platforms from the same file.
+
+===========
+Terminology
+===========
+
+There are several terms whose definition must be agreed upon in order
+to facilitate a discussion on the topic of this PEP.
+
+A *package* is something you install as a dependency and use via an
+``import`` statement. The packages on PyPI are an example of this.
+
+An *application* or *app* is an end product that other bits of code do
+not directly rely on and are not imported. Desktop applications,
+command-line tools, etc. are examples.
+
+A *lock file* records the packages that are to be installed for an
+app. Traditionally, the exact version of the package to be installed
+is specified by a lock file, but all specified packages are not always
+installed on a given platform (according a filtering logic described
+in a later section), which enables the lock file to describe
+reproducibility across multiple platforms. Examples of this are
+``package-lock.json`` from npm_, ``Poetry.lock`` from Poetry, etc.
```

Comment:
feels like we should mention here the non-standardized `requirements.txt` and perhaps elude the fact that the goal of this document is to create a standard python lock file

Reply:
This omittion is intentional because `requirements.txt` by design does not _guarantee_ to hold a complete, reproducible dependency set. It can be used as such, but it's messy to talk about it here (it's telling that this particular definition is already much longer than any other paragraph in this section).